src/CSModule.sol

```diff
@@ -999,7 +1080,19 @@ contract CSModule is
 
     /// @inheritdoc ICSModule
     function depositQueueItem(uint128 index) external view returns (Batch) {
-        return depositQueue.at(index);
+        for (uint256 p; p < _depositPriorities.length; ++p) {
+            QueueLib.Queue storage q = _getQueue(_depositPriorities[p]);
+            uint128 qLen = q.length();
+
+            if (index >= qLen) {
+                index -= qLen;
+                continue;
+            }
+
+            return q.at(index);
+        }
+
+        return Batch.wrap(0);
```

Comment:
I did not get this method. I also suggest having a queuePriority as a param here.

Reply:
It gets a view to a queue linearly. E.g. we have 2 queues [...]*50 + [...]*50, and we query index 69, which is mapped to the second queue's 19th element. We can achieve the same result off-chain, just with different sets of getters. This one is more unified.

Reply:
Still hard to digest